### ISSUE TITLE
getDescendants / getAncestors returning self is unintuitive 

### ISSUE THREAD

**Issue by jtushman** [typeorm/typeorm#3785]
https://github.com/typeorm/typeorm/blob/5d5716f8dc0d2a39c22fc70c94e271028de81846/test/functional/tree-tables/materialized-path/materialized-path.ts#L85

I was implementing this and was totally weirded out that that these functions returned itself.

for example.  if this was to model a family: a parent has many children.

it is unintuitive parent.getChildren to return parent.

What do you all think?  I know it would be a breaking change at all -- but think its much more intuitive.

At the very least we should update the docs.

**Comment by vlapo** [comment]
Logically is this behaviour really weird. One node is descendant and also ancestor for yourself.

![image](https://user-images.githubusercontent.com/2484063/54217738-7fe20400-44ec-11e9-9702-d9032726eda3.png)

**Comment by LeahCim** [comment]
> Logically is this behaviour really weird. One node is descendant and also ancestor for yourself.

No, this doesn't make sense at all 😅. How many people are their own child, or parent? And lets leave religion out of it! 🤣   ...but it probably would be a breaking change, as someone could have started relying on it. We always filter out _self_, as it never makes sense to keep it in, in our usage.

**Comment by nicu-chiciuc** [comment]
Would it make sense to add a parameter to the query so it's possible to opt into use this behaviour?
